pkg/io/confio/read_defs.go

```diff
@@ -35,46 +40,146 @@ func ReadDefs(filename string) (*ir.ConfigDefs, error) {
 		return nil, err
 	}
 
-	subnetMap := make(map[string]ir.IP)
+	instances, nifs := parseInstancesNifs(config)
+	vpes, vpeEndpoints := parseVPEs(config)
+	vpcs := parseVPCs(config)
+	err = validateVpcs(vpcs)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ir.ConfigDefs{
+		VPCs:         vpcs,
+		Subnets:      parseSubnets(config),
+		NIFs:         nifs,
+		Instances:    instances,
+		VPEEndpoints: vpeEndpoints,
+		VPEs:         vpes,
+	}, nil
+}
+
+func parseVPCs(config *configModel.ResourcesContainerModel) map[ir.ID]*ir.VPCDetails {
+	VPCs := make(map[ir.ID]*ir.VPCDetails)
+	for _, vpc := range config.VpcList {
+		addressPrefixes := make([]ir.CIDR, 0)
+		for _, addressPrefix := range vpc.AddressPrefixes {
+			addressPrefixes = append(addressPrefixes, ir.CidrFromString(*addressPrefix.CIDR))
+		}
+		VPCs[ir.ID(*vpc.Name)] = &ir.VPCDetails{AddressPrefixes: addressPrefixes}
```

Comment:
Should we prepare for the day when synthesis will be used on a multi-account setting?
Should we use CRNs all over the place instead of names (like we do in the analyzer)?
@kyorav , what do you think?

Reply:
I would prefer to use the account as a top-level scope. We wanted to avoid CRNs so that we could, one day, do the shift left thing.